### ISSUE TITLE
Changing Java Version in layman's terms for a complete idiot

### ISSUE THREAD

**Issue by MtnDrew** [BluSunrize/ImmersiveEngineering#2929]
I've never had to make Minecraft use a different version of Java before. I'm not using any particular modpack (just some random ones I've put together for myself) and am working from the standard Minecraft Launcher. I cannot for the life of me figure out how to get my launcher to use a different version of Java.

I've gone into the launch options with advanced settings and specified the javaw from the Java 8 update 161 folder. I've moved the folder into the Minecraft folder runtime subdirectory in Program Files(x86) and tried to specify the javaw from there. I've even moved the Javaw from the 161 folder to replace the one in the prepackaged Minecraft Java folder. (This went horribly, had to reinstall)

So as you can see, I'm a complete idiot when it comes to this. I've seen this github is very active, but when reading other threads about people having this problem, it seems like everyone has figured it out in the background without describing it in the thread. So if anyone could please explain it to me like I'm five, that would be so helpful. I love immersive engineering and would really like to do what I need to help it run properly.

**Comment by malte0811** [comment]
I don't know about the native launcher, in the jar launcher you can set the Java version in the profile settings IIRC (though I don't think the jar launcher has it's own Java version).
@BluSunrize it might be a good idea to make a markdown file in this repo with instructions for different launchers and put a link to it in the warning. If you agree I'll ask the guy who posted the instructions for the twitch launcher on reddit whether he'd be fine with us copying the instructions here.

**Comment by EvilDylan** [comment]
To change your java version on the vanilla Minecraft launcher, you do as follows:
First, install a newer version of Java from the proper website.
Then, select launch options on the top of the launcher:
![launcheroptions](https://user-images.githubusercontent.com/26489839/38459783-39f6313a-3a7c-11e8-84e7-f35e66c8cf0d.png)
Select the Forge profile you are currently using to edit it.
You'll see near the bottom a line that reads "Java Executable", next to a grey toggle. You must left-click this toggle to be able to edit the executable.
![javaoption](https://user-images.githubusercontent.com/26489839/38459802-96f91366-3a7c-11e8-9331-b89eba91497e.png)
As indicated above, click the folder next to the java executable line. This will open up your file explorer.
Locate your newer install of java; I assume you know where you installed it, but if not it is probably at C:\Program Files\Java\[version]\bin\
Select the file named "javaw.exe" (the extension may not show depending on your computer's settings). This should set it as the executable called in the launcher's java executable line.
Save the profile and play it, and it should now be the up-to-date version of Java.

**Comment by MtnDrew** [comment]
I guess the next step would be to figure out what I'm doing wrong

![zjava](https://user-images.githubusercontent.com/11734769/38460273-a939f30a-3a73-11e8-857e-723abe5bebb3.PNG)

After that I get this:

![zjavafail](https://user-images.githubusercontent.com/11734769/38460284-be7aec7e-3a73-11e8-8811-eb1f7ef5d25a.PNG)

**Comment by EvilDylan** [comment]
I am not quite certain what this issue could be, as those changes were the only ones I made.
It is a fresh install of jre, and does it work for other profiles?

**Comment by MtnDrew** [comment]
Ahh. Good suggestion. Vanilla Minecraft 1.12.2 launched perfectly with a switched executable. I guess it's time to review the mods I have installed.

**Comment by MtnDrew** [comment]
Alright so after moving every single mod out of the Mods folder, I still get the error. Would that mean it's the Forge version I'm using? The version is 1.12.2.14.23.2.2654. A few days ago this was the latest version of Forge.

**Comment by ProsperCraft** [comment]
I made a video on how to change to the jar launcher, and install java 64 bit because I keep running in to players with this issue - https://youtu.be/-uUr1VAnkcs  Hope it helps.
Trying the Twitch launcher might give you a different experience, who knows.

**Comment by MtnDrew** [comment]
Ah the twitch launcher. I guess I should try it. I don't know what it does differently but we can see if that works if everyone else is using it. I don't ever go on twitch so I didn't think about using it.

**Comment by mindforger** [comment]
i am using twitch launcher heavily but it ends up starting the normal launcher as in those screenshots but with predefined profiles, so you have to setit up in twitch to use proper java

**Comment by ktole1999** [comment]
looks like you are using 32bit java instead of 64bit
